src/Microsoft.Health.Fhir.Core/Features/Persistence/ResourceDeserializer.cs

```diff
@@ -38,6 +40,95 @@ public ResourceElement Deserialize(ResourceWrapper resourceWrapper)
             return resource;
         }
 
+        public ResourceElement Deserialize(RawResourceElement rawResourceElement)
+        {
+            EnsureArg.IsNotNull(rawResourceElement, nameof(rawResourceElement));
+
+            ResourceElement resource = DeserializeRaw(rawResourceElement);
+
+            return resource;
+        }
+
+        /// <summary>
+        /// Get the raw data from resourceWrapper as a string and JsonDocument.
+        /// If the RawResource's VersionSet or LastUpdatedSet are false, then the RawResource's data will be updated
+        /// to have them set to the values in the ResourceWrapper
+        /// </summary>
+        /// <param name="resourceWrapper">Input ResourceWrapper to convert to a JsonDocument</param>
+        /// <returns>A <see cref="JsonDocument"/></returns>
+        public static (string, JsonDocument) DeserializeToJsonDocument(ResourceWrapper resourceWrapper)
+        {
+            EnsureArg.IsNotNull(resourceWrapper, nameof(resourceWrapper));
+
+            if (resourceWrapper.RawResource.LastUpdatedSet && resourceWrapper.RawResource.VersionSet)
+            {
+                return (resourceWrapper.RawResource.Data, JsonDocument.Parse(resourceWrapper.RawResource.Data));
```

Comment:
In this case, we should be able to skip deserialization altogether.

Reply:
This is true for a GET on a single resource, but we need the JsonDocument for serializing it into a bundle response. I might be able to refactor so that we only create the JsonDocument when we know we need to

Reply:
Even in a bundle, we're deserializing it and serializing it untouched, right?

Reply:
Right, but when serializing the bundle using a Utf8JsonWriter, you can't write raw json and need to have a JsonDocument object.

An alternative that I got to work is to have a secondary writer on the stream (StreamWriter), and pass in the raw json for it to write out to the stream. It's a little ugly but I think it's worthwhile.

Reply:
These are the changes I mentioned: https://github.com/microsoft/fhir-server/pull/1178/files#diff-c4e157da5e4618e892bb001771725246R102-R111

It's a bit ugly (2 stream writers, and having to write the NullValue then seek back to ovewrite it to keep the Utf8JsonWriter in a happy state, and copying a memory stream to the response stream), but it does get us out of doing that extra JsonDocument.Parse